pkg/common/cns-lib/vsphere/virtualcenter.go

```diff
@@ -307,6 +311,23 @@ func (vc *VirtualCenter) connect(ctx context.Context, requestNewSession bool) er
 	}
 	// If session has expired, create a new instance.
 	log.Warnf("Creating a new client session as the existing one isn't valid or not authenticated")
+	if vc.reloadConfigForNewClient {
+		cfg, err := config.GetConfig(ctx)
+		if err != nil {
+			return logger.LogNewErrorf(log, "failed to read config. Error: %+v", err)
+		}
+		newVcenterConfigs, err := GetVirtualCenterConfigs(ctx, cfg)
+		if err != nil {
+			return logger.LogNewErrorf(log, "failed to get VirtualCenterConfigs. err=%v", err)
+		}
+		for _, newvcconfig := range newVcenterConfigs {
+			if newvcconfig.Host == vc.Config.Host {
```

Comment:
What if the user changer VC IP to FQDN? In that case we will not be able to match the new config's Host and the old host.

Reply:
In the ReloadConfiguration we do remove older entries and only retain enries with new FQDN.
if CSI needs to re-connect before ReloadConfiguration detect config changes, new connections can not be made. and we will fail connect method with following error

```golang
		if !foundVCConfig {
			return logger.LogNewErrorf(log, "failed to get vCenter config for Host: %q", vc.Config.Host)
		}
```